runtime/development/src/precompiles.rs

```diff
@@ -0,0 +1,41 @@
+use core::marker::PhantomData;
+
+use codec::Decode;
+use frame_support::dispatch::{Dispatchable, GetDispatchInfo, PostDispatchInfo};
+use pallet_evm::{Precompile, PrecompileHandle, PrecompileResult, PrecompileSet};
+use sp_core::H160;
+
+pub struct CentrifugePrecompiles<R>(PhantomData<R>);
+
+impl<R> CentrifugePrecompiles<R> {
+	#[allow(clippy::new_without_default)] // We'll never use Default and can't derive it.
+	pub fn new() -> Self {
+		Self(Default::default())
+	}
+}
+
+impl<R> PrecompileSet for CentrifugePrecompiles<R>
+where
+	R: pallet_evm::Config,
+	R::RuntimeCall: Dispatchable<PostInfo = PostDispatchInfo> + GetDispatchInfo + Decode,
+	<R::RuntimeCall as Dispatchable>::RuntimeOrigin: From<Option<R::AccountId>>,
+{
+	fn execute(&self, handle: &mut impl PrecompileHandle) -> Option<PrecompileResult> {
+		// 1025 is chosen for compatibility with Moonbeam
+		if handle.code_address() == addr(1025) {
```

Comment:
It should be a `const` anyway ;)

I'd rather not rely on `is_precompile` in `execute` since we'll need to rewrite it again to use a match on addresses when we add more precompiles in the future.

Reply:
Oh, my point is mostly about having to touch in two places (might oversee it) when updating that constant. But I do not know enough of how is the `is_precompile` used or will be used.